test/nodes/composites/selector_test.gd

```diff
@@ -125,3 +126,29 @@ func test_not_interrupt_first_after_finished() -> void:
 	
 	selector.remove_child(action3)
 
+
+func test_interrupt_when_nested() -> void:
+	var selector_reactive = auto_free(load(__selector_reactive).new())
+	var fake_condition = auto_free(load(__count_up_action).new())
+	
+	tree.remove_child(selector)
+	tree.add_child(selector_reactive)
+	selector_reactive.add_child(fake_condition)
+	selector_reactive.add_child(selector)
+	
+	fake_condition.status = BeehaveNode.FAILURE
+	action1.status = BeehaveNode.RUNNING
+	
+	assert_that(tree.tick()).is_equal(BeehaveNode.RUNNING)
+	assert_that(action1.count).is_equal(1)
+	assert_that(action2.count).is_equal(0)
+	
+	fake_condition.status = BeehaveNode.SUCCESS
+	assert_that(tree.tick()).is_equal(BeehaveNode.SUCCESS)
+	assert_that(action1.count).is_equal(0)
+	assert_that(action2.count).is_equal(0)
+	
+	# clean up...
```

Comment:
generally, individual tests should not do the cleanup but the clean up should happen in an "after_test" step

Reply:
yes but after_test would be called for every test in the file and this clean up should only happen in this particular test